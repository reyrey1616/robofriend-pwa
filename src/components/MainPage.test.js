import React from 'react';
import { shallow } from 'enzyme';
import MainPage from './MainPage';
// it('Testing the render of App Component', () => {
//   const mockStore = {
//     robots: [],
//     searchField: '',
//   };
//   expect(shallow(<MainPage store={mockStore} />)).toMatchSnapshot();
// });

let wrapper;
beforeEach(() => {
  const mockProps = {
    onRequestRobots: jest.fn(),
    robots: [],
    searchField: '',
    pending: false,
  };

  wrapper = shallow(<MainPage {...mockProps} />);
});

it('Testing the render of MainPage Component without crashing', () => {
  expect(wrapper).toMatchSnapshot();
});

it('It filters the robot correctly', () => {
  expect(wrapper.instance().filterRobots([])).toEqual([]);
});
