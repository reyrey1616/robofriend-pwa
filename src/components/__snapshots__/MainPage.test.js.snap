// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the render of MainPage Component without crashing 1`] = `ShallowWrapper {}`;
